Automattic/wp-calypso
# My Home: Remove obsolete notices location

#### Changes proposed in this Pull Request

We stopped using the notices location in #42239 / D43488-code, so this PRs removes all the dead code that is no longer used.

#### Testing instructions

Go through the site setup journey (create a site, complete site setup, launch, view first task, ...) and make sure there are no regressions (behavior should match production).
